web/src/compose_ui.ts

```diff
@@ -254,17 +282,19 @@ export function compute_placeholder_text(opts) {
         const recipient_list = opts.private_message_recipient.split(",");
         const recipient_parts = recipient_list.map((recipient) => {
             const user = people.get_by_email(recipient);
-            if (people.should_add_guest_user_indicator(user.user_id)) {
-                return $t({defaultMessage: "{name} (guest)"}, {name: user.full_name});
+            if (people.should_add_guest_user_indicator(user!.user_id)) {
+                return $t({defaultMessage: "{name} (guest)"}, {name: user!.full_name});
             }
-            return user.full_name;
+            return user!.full_name;
         });
-        const recipient_names = util.format_array_as_list(recipient_parts, "long", "conjunction");
+        const recipient_names = Intl.ListFormat
+            ? new Intl.ListFormat(user_settings.default_language).format(recipient_parts)
+            : recipient_parts.join(", ");
```

Comment:
This accidentally reverts an important change! See cd91bc3d71b5a6d37dc2695d76cabf575a565f2f. Please be more careful when rebasing to not lose changes.